packages/medusa/src/api/routes/admin/orders/create-fulfillment.ts

```diff
@@ -126,6 +128,16 @@ export default async (req, res) => {
       pvInventoryService.withTransaction(transactionManager)
 
     if (validated.location_id) {
+      const { fulfillments } = await orderService
+        .withTransaction(transactionManager)
+        .retrieve(id, {
+          relations: [
+            "fulfillments",
+            "fulfillments.items",
+            "fulfillments.items.item",
+          ],
+        })
+
       await updateInventoryAndReservations(
```

Comment:
todo: we should pass the transaction and use it in the function

Reply:
we already pass the `Tx` version of the service into the function

Reply:
```
const pvInventoryServiceTx =
        pvInventoryService.withTransaction(transactionManager)

      await updateInventoryAndReservations(
        fulfillments.filter((f) => !existingFulfillmentMap[f.id]),
        {
          inventoryService: pvInventoryServiceTx,
          locationId: validated.location_id,
        }
      )
```

Reply:
My god, i am blind.